staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -147,3 +149,39 @@ func ComputeHistogramDelta(before, after model.Samples, label model.LabelName) {
 func makeKey(a, b model.LabelValue) string {
 	return string(a) + "___" + string(b)
 }
+
+func GetMetricValueForDimension(ms Metrics, metricName, dimension string) map[string]int64 {
+	result := make(map[string]int64)
+	for method, samples := range ms {
+		if method != metricName {
+			continue
+		}
+		for _, sample := range samples {
+			count := int64(sample.Value)
+			dimensionName := string(sample.Metric[model.LabelName(dimension)])
+			result[dimensionName] = count
+		}
+	}
+	return result
+}
+
+func HasValidMetrics(metrics Metrics, metricKey string, dimensions ...string) bool {
+	var errCount int
+	log.Logf("Looking for sample in metric %q", metricKey)
+	samples, ok := metrics[metricKey]
+	if !ok {
+		log.Logf("Key %q was not found in metrics", metricKey)
```

Comment:
I'd prefer to provide common functionality in the test util package, so, I suggest removing the log print here (as well as above and below), because I don't want to involve `e2e` too much.